org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/dom/ASTConverter_23Test.java

```diff
@@ -133,4 +127,64 @@ void main() {
 		assertEquals("Not a Block", block.getNodeType(), ASTNode.BLOCK);
 		assertEquals("Block startPosition is not correct", block.getStartPosition(), 21);
 	}
+
+	public void test003_a() throws CoreException {
+		ASTParser astParser = ASTParser.newParser(getAST23());
+	    Map<String, String> options = new HashMap<>();
+	    options.put(JavaCore.COMPILER_COMPLIANCE, "23");
+	    options.put(JavaCore.COMPILER_SOURCE, "23");
+
+	    astParser.setCompilerOptions(options);
+	    astParser.setEnvironment(new String[] {}, new String[] {}, new String[] {}, true);
+	    astParser.setUnitName("Example.java");
+	    astParser.setResolveBindings(true);
+	    astParser.setBindingsRecovery(true);
+
+	    String source ="""
+		    		sealed class A permits B, C {}
```

Comment:
You might want to add some more modifiers and test for their presence in the output. For e.g.
`public sealed abstract class A permits B, C {}`

Reply:
Added new unit test to address this.